tests/test_plugin_loader.py

```
@@ -2318,6 +2554,45 @@ def test_clone_or_update_repo_checkout_fails_fallback(
+                    retry_delay=pl.GIT_RETRY_DELAY_SECONDS,
+                ),
+            ]
+            self.assertEqual(mock_run_git.call_args_list, expected_calls)
+
+    @patch("mmrelay.plugin_loader._run_git")
+    def test_try_checkout_and_pull_ref_tag_failure_returns_false_no_force_sync(
+        self, mock_run_git
+    ):
+        mock_run_git.side_effect = subprocess.CalledProcessError(1, "git checkout")
+        result = pl._try_checkout_and_pull_ref(
+            "/tmp/repo", "v1.0.0", "test-repo", ref_type="tag"
```

Ruff (0.15.7)

[error] Probable insecure usage of temporary file or directory: "/tmp/repo"

(S108)